en/getting-started/first-game/advance.md

```diff
@@ -0,0 +1,450 @@
+            }
+        }
+
+        // Generate tracks based on track type
+        let linkedBlocks = 0;
+        for (let j = 0; j < this._road.length; j++) {
+            if(this._road[j]) {
+                ++linkedBlocks;
+            }
+            if(this._road[j] == 0) {
+                if(linkedBlocks > 0) {
+                    this.spawnBlockByCount(j - 1, linkedBlocks);
+                    linkedBlocks = 0;
+                }
+            }        
+            if(this._road.length == j + 1) {
+                if(linkedBlocks > 0) {
+                    this.spawnBlockByCount(j, linkedBlocks);
+                    linkedBlocks = 0;
+                }
+            }
+        }
+    }
+
+    spawnBlockByCount(lastPos: number, count: number) {
+        let block: Node|null = this.spawnBlockByType(BlockType.BT_STONE);
+        if(block) {
+            this.node.addChild(block);
+            block?.setScale(count, 1, 1);
+            block?.setPosition(lastPos - (count - 1) * 0.5, -1.5, 0);
+        }
+    }
+    spawnBlockByType(type: BlockType) {
+        if (!this.cubePrfb) {
+            return null;
+        }
+
+        let block: Node|null = null;
+        switch(type) {
+            case BlockType.BT_STONE:
+                block = instantiate(this.cubePrfb);
+                break;
+        }
+
+        return block;
+    }
+
+    onStartButtonClicked() {
+        // To start the game by clicking the Play button
+        this.curState = GameState.GS_PLAYING;
+    }
+
+    checkResult(moveIndex: number) {
+        if (moveIndex < this.roadLength) {
+            // Jumped on the pit
+            if (this._road[moveIndex] == BlockType.BT_NONE) {
+                this.curState = GameState.GS_INIT;
+            }
+        } else {    // Jumped over the maximum length
+            this.curState = GameState.GS_INIT;
+        }
+    }
+
+    onPlayerJumpEnd(moveIndex: number) {
+        if (this.stepsLabel) {
+            // Because in the last step there may be a jump with a large pace, but at this time, whether the jump is a large pace or a small pace should not increase the score more
+            this.stepsLabel.string = '' + (moveIndex >= this.roadLength ? this.roadLength : moveIndex);
+        }
+        // Check the type of the currently falling road and get the result
+        this.checkResult(moveIndex);
+    }
+
+    // update (deltaTime: number) {
+    //     // Your update function goes here.
+    // }
+}
+```
+
+## Conclusion
+
+Congratulations on completing your first game made with Cocos Creator! You can download the full project at [GitHub](https://github.com/cocos-creator/tutorial-mind-your-step-3d) to download the complete project. We hope this quick start tutorial will help you understand the basic concepts and workflow of the Cocos Creator game development process. If you are not interested in writing and learning to script, you can also copy the script directly from the finished project.
```

Comment:
You can download the full project at [GitHub](https://github.com/cocos-creator/tutorial-mind-your-step-3d) **to download the complete project**    the bold words should be removed.